### PR TITLE
#rWXS2NYV Fixing the setting up of the environment

### PR DESCRIPTION
**Pull request by Ally4**
#### What does this PR do?
This PR sets up node.js rest api empty project.
#### Description of Task to be completed?
- Setting up the project structure
- Setting up a welcome endpoint
- Configuring ESLint for code analysis
- Setting up the api documentation with swagger
#### How should this be manually tested?
- By cloning the repository
- Running in the terminal `npm install`
- Running `npm run dev`
- Checking the `http://localhost:4000/` in the postman
- Checking `http://localhost:4000/api-docs/#/` for the swagger documentation
#### Any background context you want to provide?
- Installation of packages
#### What are the relevant Trello id stories?
#rWXS2NYV

### CODE REVIEW

**Review by Mugema123 — DISMISSED** [review]
Everything is working correctly in postman.

File: package.json [hunk 1]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "eagles-phantom-backend",
   "version": "1.0.0",
-  "description": "",
   "main": "index.js",
   "scripts": {
-    "dev": "babel-node server/app.js"
+    "dev": "cross-env NODE_ENV=development nodemon --exec babel-node ./server/app.js"
```
Comment (Ntwali-Josue) [inline package.json]: why do you have two devs in your scripts?try to use one

**Review by Ntwali-Josue — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Ally4) [reply package.json]: The first dev is no longer in the code, that - shows that it has been removed.

**Review by Ntwali-Josue — DISMISSED** [review]

**Review by TuyizeeAnastase — DISMISSED** [review]

**Review by Mugema123 — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (blaise82) [inline package.json]: you don't need cross-env to set a NODE_ENV example: `NODE_ENV=test` will work fine
on a start dev server command you don't need to specify the NODE_ENV

File: server/app.js [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
 app.use(express.json());
 
 app.use(bodyParser.json());
 
+app.use('/', routes);
+
+app.use(cors());
+
+
+app.use("/api-docs", swaggerUI.serve, swaggerUI.setup(swagger));
```
Comment (blaise82) [inline server/app.js]: don't put any route in your app.js file
so all I expect is `app.use(router); ` in your app.js file

File: server/app.js [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+
+
 const port = process.env.PORT || 4000;
 
 app.listen(port, () => {
-    console.log(`This is where the port is running from ... ${port} `)
+  console.log(`The server is running on ${port}`);
 });
 
 export default app;
```
Comment (blaise82) [inline server/app.js]: add a new line

File: swagger.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (blaise82) [inline swagger.json:1]: make sure swagger detects the root URL too
so that pipelines might be tested easily

File: server/controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const welcome = (req, res) => {
+      return res.status(200).json({
+          status: 200,
+          message: 'Welcome to phantom, a platform to facilitate the transportation mode in town!.'
+      });
+  };
+
+  export default welcome;
```
Comment (blaise82) [inline server/controllers/user.js]: instal eslint on your vscode so that new lines can be added to every end of your file

File: swagger.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                "description": "Landing page",
+                "responses": {
+                    "200": {
+                        "description": "Welcome to phantom"
+                    }
+                }
+            }
+        }
+}
+}
```
Comment (blaise82) [inline swagger.json:31]: add a new line

**Review by Ntwali-Josue — DISMISSED** [review]

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Ally4) [reply package.json]: Thanks, its solved

**Review by TuyizeeAnastase — DISMISSED** [review]

**Review by Ntwali-Josue — DISMISSED** [review]

**Pull request closed without merging**